include/aspect/compat.h

```diff
@@ -83,6 +83,67 @@ namespace dealii
 }
 #endif
 
+/*
+ * Unique_ptr functionality that was introduced in deal.II 8.3 and replaces
+ * auto_ptr in CXX17. We need this to silence deprecation warnings in new
+ * compilers.
+ */
+#if !DEAL_II_VERSION_GTE(8,3,0)
```

Comment:
Why do you need this check for the deal.II version? Are you including some header that in turn includes the deal.II std_cxx11/unique_ptr.h?